scripts/x-api.js

```diff
@@ -0,0 +1,596 @@
+/** @namespace H5P */
+H5P.VideoXAPI = (function ($) {
+
+  /**
+   * Xapi video statement generator for H5P.
+   *
+   * @class
+   * @param {Object} instance Parent H5P.Video{YouTube|Html5|Flash}
+   *                          video object generating xAPI statements
+   */
+  function XAPI(instance) {
+    var self = this;
+
+    /**
+     * Variables to track time values from the video player.
+     *
+     * @public
+     */
+    self.previousTime = 0;
+    self.seeking = false;
+    self.seekedTo = 0;
+    self.duration = 0;
+
+    /**
+     * Variables to track internal video state.
+     *
+     * @private
+     */
+    var videoInstance = instance;
+    var seekStart = null;
+    var playedSegments = [];
+    var playedSegmentsSegmentStart =0;
+    var playedSegmentsSegmentEnd;
+    var volumeChangedOn = null;
+    var volumeChangedAt = 0;
+    var sessionID = guid();
+    var currentTime = 0;
+
+
+    /**
+     * Generates "initialized" xAPI statement (Video Profile).
+     * @see https://liveaspankaj.gitbooks.io/xapi-video-profile/content/statement_data_model.html#231-initialized
+     *
+     * @public
+     * @param {Number} currentTime time of the video currently
+     * @param {Number} width width of the current screen
+     * @param {Number} height height of the current video screen
+     * @param {Number} rate playback rate
+     * @param {number} volume level of volume
+     * @param {Boolean} ccEnabled boolean whether closed captions are enabled
+     * @param {String} ccLanguage language of closed captions
+     * @param {String} quality quality rating of resolution
+     * @returns {Object} JSON xAPI statement
+     *
+     */
+    self.getArgsXAPIInitialized = function (currentTime, width, height, rate, volume, ccEnabled, ccLanguage, quality) {
+      // Set default value for quality.
+      quality = typeof quality !== 'undefined' ? quality : Math.min(width, height);
+
+      // Variables used in compiling xAPI results.
+      var dateTime = new Date();
+      var timeStamp = dateTime.toISOString();
+      var resultExtTime = formatFloat(currentTime);
+      var screenSize = screen.width + "x" + screen.height;
+      var playbackSize = (width !== undefined && width !== '' ) ? width + "x" + height : "undetermined";
+      var playbackRate = rate;
+      var volume = formatFloat(volume);
+      var userAgent = navigator.userAgent;
+      var state = document.fullScreen || document.mozFullScreen || document.webkitIsFullScreen || false;
+
+      var extensions = {};
+      if (typeof state !== "undefined" && state != false) {
+        extensions["https://w3id.org/xapi/video/extensions/full-screen"] = state
```

Comment:
I mean, it's a URL, is it worth refactoring out common components? Should we have a constant for `https://` as well? I think it reads easier keeping the URLs as complete strings, instead of concatenated pieces.